corehq/apps/domain/utils.py

```diff
@@ -124,3 +124,21 @@ def get_serializable_wire_invoice_general_credit(general_credit):
         }]
 
     return []
+
+
+def get_dte_expressions(domain_name):
```

Comment:
To ensure that things work consistently with Solutions, we are allowing all the expressions by default. We would be restricting certain expressions for SaaS partners manually. So that is why it is returning a full list of all restricted expressions by default.

> I would think it should return an empty list if the model doesn't exist. In this case, I'd also recommend renaming it to 'get_allowed_ucr_expressions'

I will update the name.

Reply:
It contains the list of allowed UCR expressions for a domain.

> What is the logic to get the list of restricted expressions for a domain?

First thing is that the change only affects two expressions `BaseItemExpression` and `RelatedDocumentExpressions`.  If a user tries to create any of the above expressions, we will see if the domain is allowed to use the expression. For that, we will query `AllowedDTEExpressionSettings`. If the expression is present in the model then we will allow it else we will raise an error.

Reply:
OK, I think it's making sense now. I think it would actually be better to store the list of disallowed expressions in the DB. The reason I think that is because it makes it easier to add more expressions to the list in future without having to do a migration.

Reply:
The reason we are saving only allowed attributes is because -
- It translates well with the UI as we have to manually add the allowed expressions to the list.
- The list is unlikely to grow in the future, the expressions that are being restricted seemed to have a platform. Where abusing the expression could have caused performance issues. We can work on making them GA in the future but things going other way round is highly unlikely.